Enhance places by events/surname with list of individuals (fix #738)

hgouraud opened the pull request:
* limit between long/short controlled by bvar.nb_places_short. Default is 30
* possible to switch between short/long with button at beginning
* selecting a place displays that place in short/long format. Button to alternate is name of the place.

sagotch reviewed (changes requested):
J'ai fais quelque commentaires.

J'ai l'impression qu'il y a deux chose dans cette PR. Un fix + une nouvelle feature, que je ne comprend pas.

Je voudrais bien être éclairé sur cet ajout :)

hgouraud commented:
Non à mon avis. Il faudrait passer toute la liste en paramètre!! 
Je vois pas de solution immédiate!!
Si tu as une piste, je veux bien essayer de l’explorer. 

Envoyé de mon iPhone

> Le 18 janv. 2019 à 12:04, Julien Sagot <notifications@github.com> a écrit :
> 
> @sagotch commented on this pull request.
> 
> In lib/place.ml:
> 
> > @@ -153,6 +153,23 @@ let get_opt conf =
>    (if add_burial then "&bu=on" else "") ^
>    (if add_marriage then "&ma=on" else "")
>  
> +let sort_button conf =
> Ça serait possible de le faire dans le template ou pas ?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

a2line commented:
Additions:
* the count of persons by surname by events/place “Dupont (30)”
* the total of persons for all surname by events/place “Rouen (1408)”
* a form with inputs and button on `m=RL` and `m=L` (template is `buttons_places.txt`)
  * add two new inputs for variables Place= and Surname=
  * add a button to sort by frequency instead of alphabeticaly
* links on counts send to the list `m=L` of individuals by surname by events/place (template is `list.txt`)
  * a short table of events is linked to each individual (french notation BMS/NMD is used).
  * add a multi-relation graph `m=RLM` to draw the tree linking all individuals by surname by events/place (the events are kept on the graph using t#= description).

hgouraud committed:
Allow loop for substrings of evar.s with tail/cnt/is_first and UTF-8 support

a2line committed:
Enhance navigation on books, remove too heavy js, add link to Wikidata

hgouraud committed:
Display places in long/short form with new features (fix #738)

Propose search results for exact/substring

a2line committed:
Add two templates files: list of individuals by place, second one for its buttons

Alphabetical sort of the list by default, extra button to sort by dates.
Two more buttons to show/hide dates or parents.
First column shows events (using NMD/BMS).
Link to graph m=RLM of all listed individual, sometime non-pertinent.
Add links to book of place for easier renaming.

sagotch commented:
Please do not remove the ` work-in-progress ` tag until this branch is really stable. You keep pushing commits everyday. If it is not ready yet, let's finish it before merging something half-working.

a2line commented:
It's always ready in a way. If not sure, if it's “half-working” or not, just ask. No PR modification is perfect crystal clear for centuries and won't be modified after a merge to the master. We just keep adding things here to simplify work operations between ourselves because we have other PR (and a big one) opened and we don't want to multiply working branches (and redo some file modifications we already did because we forget them or “need” them here). It's easier to work together like this. It was a quick fix, we add features and features, rebase on master, remove some commits for another PR to clean things for reviewer, or rebase -i to clean again because of mixup between our respective commits (that was done three or four times), time goes on. The more you wait the better it is, but the more we loose time on finition instead of fixing other issues.

Henri is ok with last rebase/state of the art. If the two commits on .ml are OK, I'll rebase a last time merging the .ml with associated template files. I think they would be better on one commit each (place.ml / upddata.ml) but I splitted in 4 for an easier rebase -i of the 20 commits of yesterday.

GuillaumeBrochu commented:
I have started to test. 

First impression of the new feature :+1: :+1: :+1: :1st_place_medal:

GuillaumeBrochu commented:
Few other comments for tonight :

- Will you eventually translate the letter code from "NBMDS" in French to "BBMDB" in English (although there are a lot of Bs in English...) and to other languages? Another option would be to use "BPMDR" to refer to the V6 standard personal event letters in .gw files (https://geneweb.tuxfamily.org/wiki/gw#Personal_information)
- Book of places (using the magic wand) give me a very strange result (**see picture below** ... place name built letter by letter... )
- For the check-boxes : I will not use "Spouse" for the English translation of "Marriage" in French
- List of individuals by place : displaying the occurrence number like [1] [2] may be confusing (it took me some time to find why there are numbers next to some people but not all of them...)
- Multi relations graph : WOW!!!  :+1: 


![screenshot_20190129_bookofplaces](https://user-images.githubusercontent.com/11491888/51956695-aa816d00-2416-11e9-9127-b4705f0613dc.png)

a2line commented:
Thanks Guillaume.
* Translation of NMDBS has been discussed a bit, it can be done but it's quite hard to choose. First idea was a small table with “X XX " but it was far harder to read. We can perhaps do something for English at last. We chose French because of redundant letters in English, and we don't have good icons for events. We can perhaps add a abbr/title on the French letters to do the full translation of what is what?
* The magic wand has been added at the beginning as a bridge to help to quickly modify a place entry in the book of places, currently the breadcrumb prints all the substrings of s= and it's too much as you noticed, I agree with that, but we don't have a function to have a substring of the name of the place on lists so can’t use a shorter &s=… for the moment (4 or 5 letters would be enough/better). We can also remove that magic wand now that the book has been a bit enhanced too to find places. To be improved!
* Translation of evar.ma checkbox is clearly wrong!!
* Occurrences were in brackets because of “(dates)” in parentheses (to avoid double parentheses), but we reverted the removal of the extra space before all short dates yesterday (too much used in template files for the moment), so we can now use parentheses for occurrences, good idea « Marie Durand (1) 1874-1943 ».
* m=RLM is amazing for one surname/place; all individuals/place is sometimes good but often leads to strange graphs.

Thanks again for that very good feedback!

hgouraud commented:
The current solution displaying all substrings of the place is indeed disturbing. What would be a good alternative? The last n (where n could be defined in the config file, default 3)?
Marie (1) Durand 1874-1943 might be more explicit?


> Le 30 janv. 2019 à 07:19, A2 <notifications@github.com> a écrit :
> 
> Thanks Guillaume.
> 
> 	• Translation of NMDBS has been discussed a bit, it can be done but it's quite hard to choose. First idea was a small table with “X XX " but it was far harder to read. We can perhaps do something for English at last. We choose French because of redundant letters in English, and we don't have good icons for events. We can perhaps add a abbr/title on the French letters to do the full translation of what is what?
> 	• The magic wand has been added at the beginning as a bridge to help to quickly modify a place entry in the book of places, currently the breadcrumb prints all the substrings of s= and it's too much as you noticed, I agree with that, but we don't have a function to have a substring of the name of the place on lists so can’t use a shorter &s=… for the moment (4 or 5 letters would be enough/better). We can also remove that magic wand now that the book has been a bit enhanced too to find places. To be improved!
> 	• Translation of evar.ma checkbox is clearly wrong!!
> 	• Occurrences were in brackets because of “(dates)” in parentheses (to avoid double parentheses), but we reverted the removal of the extra space before all short dates yesterday (to much used in template files for the moment), so we can now use parentheses for occurrences, good idea « Marie Durand (1) 1874-1943 ».
> 	• m=RLM is amazing for one surname/place; all individuals/place is sometimes good but often leads to strange graphs.
> Thanks again for that very good feedback!
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

hgouraud committed:
Proposal to reduce the number of substrings in foreach.substr

hgouraud committed:
Fix maxs when s=. %evar.n.s; gives the n first cars of variable s;

hgouraud committed:
Trying A > Ab > ... > Abcdef > ...

hgouraud committed:
Invert elements before sort in dictionnary

hgouraud committed:
Cleaner k1/k2, k3 is full location

hgouraud committed:
k3 in reverse order

a2line commented:
Back to WIP to digest the feedbacks, and also fix a big forgetting.

a2line committed:
explicit NMDBS, fix translation, remove RLM/sort if nb=1…

GuillaumeBrochu commented:
Is it really necessary to **display** the occurrence number in this page? This is not genealogy data... this is "under the hood" database-related info IMO! People that are not advanced users will be confused by the meaning of this number.

And for the NMBDS display, we could simply use the code that is already used for "old V6" .gw files : [#bp PlaceOfBirth] [#pp BaptizePlace] [#mp WeddingPlace] [#dp PlaceOfDeath] [#rp BurialPlace] -> BPMDR and use a legend at the [top/bottom] of the page that is using the translated lexicon?

a2line commented:
A simple idea I already had to explict the NMDBS initials was to add them after the list of events in the summary/header.

`Events: birth N, baptism B, marriage M, death D, burial S.`

But I though it was ruining readability (perhaps because it was repetitive for French) and reverted that. What do you think of this?

a2line commented:
Didn't saw your message: I just removed the occurence number after trying to place it between first_name/surname, or in a column before (events were further so harder to read). Rip.

a2line commented:
BPMDR is confusing, nobody uses that, like occurrences those are internal technical choices and I don’t like them at all.

Edit: I also agree this NMD/BMS choice is a mess for non-French users… but the solution is not trivial. An English oriented choice will also be confusing for other language-speaking users. If we take the initials of the translation systematically, it will lead to redundant letters in various languages and that is problematic on RLM graph where we don’t have the display in fake-columns (monospace font).

GuillaumeBrochu commented:
I think using a legend is a good idea, and it gives more flexibility for the letter code because it will be explicitly defined in the same page.

a2line commented:
OK. Trying this again, I remember another argument of why I removed it: some people can have NMBDS on a place even if we ask for fewer type of events, and it's silly not to print them. The only solution I see for that for the moment is to systematically print all the possible events with its initial after, then play with colors (blue or gray), like this:

![image](https://user-images.githubusercontent.com/10353895/52103410-d0a93780-25e5-11e9-966b-124efbb8d4a6.png)

hgouraud committed:
k4 to handle [xxx] - aaa, bbb k4=aaa

hgouraud committed:
Fix substr_start to work with utf-8

a2line committed:
fix evar.n.s =substr_start, support endash and emdash (wip)

hgouraud committed:
max value of cnt dans foreach initial, entry and value

a2line committed:
try compact version of substring for long s

hgouraud committed:
nbs=3 in welcome.txt, place.ml and list.txt. Slight change in upddata.txt

hgouraud committed:
Remov trace in list.txt. Adjust upddata.txt. %count in updateData.ml

hgouraud committed:
Test for git presence in compilation.ml

hgouraud committed:
Replace etst for git by reset of date parameter

hgouraud committed:
k3 is now exactly the original place

hgouraud committed:
Effort to frequency sort long display

hgouraud committed:
More efforts at freq_sort

hgouraud committed:
Fixed freq_sort in long mode

hgouraud committed:
New dates_sort without precision for proper sorting

hgouraud commented:
This PR is ready for review/merge

sagotch commented:
C'est noté.

sagotch reviewed (changes requested):
Je n'ai pas testé ni vraiment regardé le fonctionement de la nouvelle fonctionnalité, mais j'ai déjà fait quelques remarques qui m'on sautés aux yeux en parcourant la PR.

Je regarderais la partie fonctionnelle plus tard, mais il y a déjà matière à corrections.

hgouraud committed:
handle places with ()

hgouraud committed:
replace quote_escaped by escape_html

hgouraud committed:
Adjust test_place to effective code

hgouraud committed:
Handle en and em dash for suburb definitions in dict and places

hgouraud committed:
More efficient replacement of em/en dash (I hope)

hgouraud committed:
Better regular expression for em/en dash

hgouraud committed:
remove localize from place.ml. Scan place for [ ] and -

hgouraud committed:
Revert "remove localize from place.ml. Scan place for [ ] and -"

This reverts commit f64b7e4c11cd7e000b2919a1a78a14c1b8b69974.

hgouraud committed:
Keep normalize in place.ml for local use. Invert unfold_places_long

hgouraud committed:
remove - in location names if not exact

hgouraud committed:
Do not compute RLM graph if nb > max_rlm_nbr (in parameter file)

hgouraud committed:
Simplify place.ml, use Templ functions

dominique95880 commented:
Bonjour Henri

Je rencontre une erreur de compilation en voulant essayer cette nouvelle version
Voici ce qu'indique la console (sous windows 10)
`Microsoft Windows [version 10.0.17763.475]
(c) 2018 Microsoft Corporation. Tous droits réservés.

09/05/2019 13:06:14,24 Microsoft Windows [version 10.0.17763.475] C:\
C>cygwin

adm0018@Derdo01 ~
$ cd ..

adm0018@Derdo01 /home
$ mkdir hgouraudtest

adm0018@Derdo01 /home
$ cd hgouraudtest

adm0018@Derdo01 /home/hgouraudtest
$ git clone https://github.com/hgouraud/geneweb.git
Clonage dans 'geneweb'...
remote: Enumerating objects: 163, done.
remote: Counting objects: 100% (163/163), done.
remote: Compressing objects: 100% (105/105), done.
remote: Total 68418 (delta 95), reused 94 (delta 58), pack-reused 68255                   100%
Réception d'objets: 100% (68418/68418), 41.60 MiB | 10.05 MiB/s, fait.
Résolution des deltas: 100% (47959/47959), fait.
Extraction des fichiers: 100% (813/813), fait.

adm0018@Derdo01 /home/hgouraudtest
$ cd geneweb

adm0018@Derdo01 /home/hgouraudtest/geneweb
$ git checkout place1
La branche 'place1' est paramétrée pour suivre la branche distante 'place1' depuis 'origin'.
Basculement sur la nouvelle branche 'place1'

adm0018@Derdo01 /home/hgouraudtest/geneweb
$ ./configure
[INFO] Checking OS type ......................................... WINDOWS

adm0018@Derdo01 /home/hgouraudtest/geneweb
$ make distrib
echo "GeneWeb [*version][:] %version; compiled on " > hd/etc/version.txt
echo "$(date '+%Y-%m-%d')" >> hd/etc/version.txt
echo " from commit " >> hd/etc/version.txt
echo "$(git show -s --date=short --pretty=format:'<a href="https://github.com/geneweb/geneweb/commit/%h">%h (%cd)</a>')" >> hd/etc/version.txt
echo "let prefix =" > lib/gwlib.ml
echo "  try Sys.getenv \"GWPREFIX\"" >> lib/gwlib.ml
echo "  with Not_found -> \"/usr\"" | sed -e 's|\\|/|g' >> lib/gwlib.ml
Generating bin/distrib/ged2gwb/ged2gwb.ml... Done!
Generating bin/distrib/ged2gwb/ged2gwb2.ml... Done!
Generating bin/distrib/setup/setup.ml... Done!
Generating lib/templ.ml... Done!
Generating lib/update.ml... Done!
sed -e "s/%%%API%%%//g" -e "s/%%%API_DEP%%%//g" lib/dune.in > lib/dune
dune build bin/distrib/connex.exe bin/distrib/ged2gwb.exe bin/distrib/ged2gwb2.exe bin/distrib/gwb2ged.exe bin/distrib/gwc1.exe bin/distrib/gwc2.exe bin/distrib/gwd.exe bin/distrib/gwdiff.exe bin/distrib/gwtp.exe bin/distrib/gwu.exe bin/distrib/mk_consang.exe bin/distrib/setup.exe bin/distrib/update_nldb.exe
[0;1;31m      ocamlc[0m lib/.geneweb.objs/byte/geneweb__Place.{cmi,cmo,cmt} [0;1;31m(exit 2)[0m
[0;2;37m(cd _build/default && C:\cygwin64\home\adm0018\.opam\ocaml-variants.4.07.1+mingw64c\bin\ocamlc.opt.exe -w @a-4-29-40-41-42-44-45-48-58-59-60-40 -strict-sequence -strict-formats -short-paths -keep-locs -w +a-4-9-35-44-48 -g -bin-annot -I lib/.geneweb.objs/byte -I C:/cygwin64/home/adm0018/.opam/ocaml-variants.4.07.1+mingw64c/lib/camlp5 -I C:\\cygwin64\\home\\adm0018\\.opam\\ocaml-variants.4.07.1+mingw64c\lib\bisect_ppx\runtime -I C:\\cygwin64\\home\\adm0018\\.opam\\ocaml-variants.4.07.1+mingw64c\lib\bytes -I C:\\cygwin64\\home\\adm0018\\.opam\\ocaml-variants.4.07.1+mingw64c\lib\markup -I C:\\cygwin64\\home\\adm0018\\.opam\\ocaml-variants.4.07.1+mingw64c\lib\uchar -I C:\\cygwin64\\home\\adm0018\\.opam\\ocaml-variants.4.07.1+mingw64c\lib\uutf -I lib/wserver/.wserver.objs/byte -I lib/wserver/.wserver.objs/native -no-alias-deps -opaque -open Geneweb -o lib/.geneweb.objs/byte/geneweb__Place.cmo -c -impl lib/place.pp.ml)[0m
File "lib/place.ml", line 584, characters 36-40:
Error: This expression has type string but an expression was expected of type
         bytes
make: *** [Makefile:167: install-exe] Error 1

adm0018@Derdo01 /home/hgouraudtest/geneweb
$`
§ en question (lib/place.ml lignes  581-587)
let str_replace str c1 c2 =
  let bstr = Bytes.of_string str in
  let rec loop bstr i =
    if bstr.[i] = c1 then Bytes.set bstr i c2;
    if i < (Bytes.length bstr - 1) then loop bstr (i+1)
  in loop bstr 0;
  Bytes.to_string bstr

Dominique

hgouraud commented:
Très étrange! ça compile (et marche) bien chez moi (Mac). commit 0c0cb5526.
Julien!! une idée de ce qui se passe?

sagotch commented:
> Error: This expression has type string but an expression was expected of type bytes

Normalement une version trop ancienne de OCaml mais je vois `ocaml-variants.4.07.1` donc peut être pas.

Essaye un
```
eval `opam config env`
```

sagotch commented:
Ah quoi que, j'avais pas lu le code.

`if bstr.[i] = c1 then Bytes.set bstr i c2;` -> `if Bytes.get bstr i = c1 then Bytes.set bstr i c2;` ?

dominique95880 commented:
Bonjour Julien
Une compilation qui se termine bien (branche master de geneweb/geneweb) fournit un fichier commit.txt comme celui-ci :
```
Commit: 70476a93, mer.  8 mai 2019 11:56:33
The OCaml toplevel, version 4.07.1
Camlp5 version 7.07-exp (ocaml 4.07.1)
```

Je ne pense pas que le compilateur soit trop vieux (je l'ai obtenu par le fichier .bat qu'A2line a mis en place tout récemment.
Dominique

sagotch commented:
Effectivement, travis remonte la meme erreur : https://travis-ci.org/geneweb/geneweb/jobs/529951697

À partir de 4.06, String et Bytes ne sont plus compatibles.

Il faut donc corriger le code effectivement.

hgouraud committed:
Fix bytes/string incompatibility

hgouraud commented:
Oui, je suis encore en 4.05!! Je migre ev 4.06 (ou 07) asap
Poussé la correction!

dominique95880 commented:
Bon pour moi.

dominique95880 commented:
Bonjour Henri

J'ai parlé trop vite car gwd plante lorsque je fait ceci avec la version de geneweb "référencée" par 
GeneWeb Version : 7.00-exp compiled on 2019-05-09 from commit 2841952f8 (2019-05-09)

J'utilise la base grimaldi (qui vient d'être supprimée pour une raison que j'ignore) et je fais les requêtes successives suivantes : 
1° Sélection de Mariage et clic sur search : OK
J'ai la liste des lieux de mariages de la base grimaldi (en mode visiteur) soit
ALLEMAGNE (2), BELGIQUE (2), EIRE (4), FRANCE (16), GERMANY (12), MONACO (2), UNITED STATES (4), UNITED STATES OF AMERICA (2)

2° Clic sur Monaco : KO presque immédiat
L'URL est alors http://localhost:2317/grimaldi?m=PS&ma=on&dates=on&k1=MONACO&long=on et Firefox affiche la page "La connexion a été réinitialisée".

La console gwd affiche alors ceci : 
GeneWeb 7.00-exp - Possible addresses:
http://localhost:2317/base
http://127.0.0.1:2317/base
http://Derdo01:2317/base
where "base" is the name of the database
Type control C to stop the service
Ready 2019-05-10 13:11 port 2317...
2019-05-10 13:11:16 (196) grimaldi?
  From: 127.0.0.1
  Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:66.0) Gecko/20100101 Firefox/66.0
  #accesses 1 (#welcome 1) since 10/05/2019
2019-05-10 13:11:29 (196) grimaldi?m=PS&ma=on
  From: 127.0.0.1
  Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:66.0) Gecko/20100101 Firefox/66.0
  Referer: http://localhost:2317/grimaldi
  #accesses 2 (#welcome 1) since 10/05/2019
2019-05-10 13:11:37 (204) grimaldi?m=PS&ma=on&dates=on&k1=MONACO&long=on
  From: 127.0.0.1
  Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:66.0) Gecko/20100101 Firefox/66.0
  Referer: http://localhost:2317/grimaldi?m=PS&ma=on
  #accesses 3 (#welcome 1) since 10/05/2019
Wserver: uncaught exception: Invalid_argument("index out of bounds")
2019-05-10 13:11:37 (208) grimaldi?m=PS&ma=on&dates=on&k1=MONACO&long=on
  From: 127.0.0.1
  Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:66.0) Gecko/20100101 Firefox/66.0
  Referer: http://localhost:2317/grimaldi?m=PS&ma=on
  #accesses 4 (#welcome 1) since 10/05/2019
Wserver: uncaught exception: Invalid_argument("index out of bounds")
2019-05-10 13:11:37 (196) grimaldi?m=PS&ma=on&dates=on&k1=MONACO&long=on
  From: 127.0.0.1
  Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:66.0) Gecko/20100101 Firefox/66.0
  Referer: http://localhost:2317/grimaldi?m=PS&ma=on
  #accesses 5 (#welcome 1) since 10/05/2019
Wserver: uncaught exception: Invalid_argument("index out of bounds")
2019-05-10 13:11:37 (196) grimaldi?m=PS&ma=on&dates=on&k1=MONACO&long=on
  From: 127.0.0.1
  Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:66.0) Gecko/20100101 Firefox/66.0
  Referer: http://localhost:2317/grimaldi?m=PS&ma=on
  #accesses 6 (#welcome 1) since 10/05/2019
Wserver: uncaught exception: Invalid_argument("index out of bounds")
2019-05-10 13:11:37 (208) grimaldi?m=PS&ma=on&dates=on&k1=MONACO&long=on
  From: 127.0.0.1
  Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:66.0) Gecko/20100101 Firefox/66.0
  Referer: http://localhost:2317/grimaldi?m=PS&ma=on
  #accesses 7 (#welcome 1) since 10/05/2019
Wserver: uncaught exception: Invalid_argument("index out of bounds")
2019-05-10 13:11:37 (200) grimaldi?m=PS&ma=on&dates=on&k1=MONACO&long=on
  From: 127.0.0.1
  Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:66.0) Gecko/20100101 Firefox/66.0
  Referer: http://localhost:2317/grimaldi?m=PS&ma=on
  #accesses 8 (#welcome 1) since 10/05/2019
Wserver: uncaught exception: Invalid_argument("index out of bounds")
2019-05-10 13:11:37 (196) grimaldi?m=PS&ma=on&dates=on&k1=MONACO&long=on
  From: 127.0.0.1
  Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:66.0) Gecko/20100101 Firefox/66.0
  Referer: http://localhost:2317/grimaldi?m=PS&ma=on
  #accesses 9 (#welcome 1) since 10/05/2019
Wserver: uncaught exception: Invalid_argument("index out of bounds")
2019-05-10 13:11:37 (196) grimaldi?m=PS&ma=on&dates=on&k1=MONACO&long=on
  From: 127.0.0.1
  Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:66.0) Gecko/20100101 Firefox/66.0
  Referer: http://localhost:2317/grimaldi?m=PS&ma=on
  #accesses 10 (#welcome 1) since 10/05/2019
Wserver: uncaught exception: Invalid_argument("index out of bounds")
2019-05-10 13:11:37 (204) grimaldi?m=PS&ma=on&dates=on&k1=MONACO&long=on
  From: 127.0.0.1
  Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:66.0) Gecko/20100101 Firefox/66.0
  Referer: http://localhost:2317/grimaldi?m=PS&ma=on
  #accesses 11 (#welcome 1) since 10/05/2019
Wserver: uncaught exception: Invalid_argument("index out of bounds")
2019-05-10 13:11:37 (204) grimaldi?m=PS&ma=on&dates=on&k1=MONACO&long=on
  From: 127.0.0.1
  Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:66.0) Gecko/20100101 Firefox/66.0
  Referer: http://localhost:2317/grimaldi?m=PS&ma=on
  #accesses 12 (#welcome 1) since 10/05/2019
Wserver: uncaught exception: Invalid_argument("index out of bounds")

Les accès n° 3 à 12 (soit 10 en tout) plantent avec le message "index out of bounds"

Faites vous le même constat ?

Dominique

hgouraud commented:
Corrigé!

hgouraud commented:
Quelqu'un sait il pourquoi la requête en erreur est répétée 10 fois?

dominique95880 commented:
Après application de la correction, ça passe
GeneWeb Version : 7.00-exp compiled on 2019-05-10 from commit a3daf08e4 (2019-05-10)

Rien à voir avec "place1", mais j'ai souvent et depuis longtemps des messages comme celui ci dans la console : "read" failed: Une connexion établie a été abandonnée par un logiciel de votre ordinateur hôte.

GeneWeb 7.00-exp - Possible addresses:
http://localhost:2317/base
http://127.0.0.1:2317/base
http://Derdo01:2317/base
where "base" is the name of the database
Type control C to stop the service
Ready 2019-05-10 14:23 port 2317...
2019-05-10 14:23:45 (196) ?
  From: 127.0.0.1
"read" failed: Une connexion établie a été abandonnée par un logiciel de votre ordinateur hôte.

2019-05-10 14:23:45 (196) ?
  From: 127.0.0.1
"read" failed: Une connexion établie a été abandonnée par un logiciel de votre ordinateur hôte.

2019-05-10 14:23:46 (196) ?
  From: 127.0.0.1
"read" failed: Une connexion établie a été abandonnée par un logiciel de votre ordinateur hôte.

2019-05-10 14:23:47 (204) grimaldi?
  From: 127.0.0.1
  Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:66.0) Gecko/20100101 Firefox/66.0
  #accesses 1 (#welcome 1) since 10/05/2019
"read" failed: Une connexion établie a été abandonnée par un logiciel de votre ordinateur hôte.

"read" failed: Une connexion établie a été abandonnée par un logiciel de votre ordinateur hôte.

2019-05-10 14:23:55 (204) grimaldi?m=PS&ma=on
  From: 127.0.0.1
  Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:66.0) Gecko/20100101 Firefox/66.0
  Referer: http://localhost:2317/grimaldi
  #accesses 2 (#welcome 1) since 10/05/2019
2019-05-10 14:23:57 (208) grimaldi?m=PS&ma=on&dates=on&k1=MONACO&long=on
  From: 127.0.0.1
  Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:66.0) Gecko/20100101 Firefox/66.0
  Referer: http://localhost:2317/grimaldi?m=PS&ma=on
  #accesses 3 (#welcome 1) since 10/05/2019

P.S. Pour cette log, j'ai effacé au préalable le fichier grimaldi.txt du répertoire cnt, ce qui fait que je retourne au message #1 et suivants.

lledieu commented:
Bonjour,

> 2019-05-10 14:23:46 (196) ?
From: 127.0.0.1
"read" failed: Une connexion établie a été abandonnée par un logiciel de votre ordinateur hôte.

J'ai également ces connexions "fantômes" que j'aimerais bien comprendre. Je pense que c'est Firefox qui prend l'initiative d’envoyer des requêtes avec un header spécial. L'erreur est spécifique Windows (sous Linux, pas d'erreur). Il faudra qu'on trouve moyen de comprend à quoi ça correspond pour filtrer ces requêtes ce qui évitera d'avoir ça dans les logs.

sagotch commented:
> J'utilise la base grimaldi (qui vient d'être supprimée pour une raison que j'ignore)

Le dossier contenant grimaldi contenait des fichiers qui n'auraient pas du être inclus dans le repo. Ce dossier "servait" de test pour l'API mais n'était de toute façon pas utilisé. grimaldi.ged n'était pas visé par cette suppression mais étant donné qu'il n'était utilisé nulle part ailleur, il a sauté avec le reste.

Si on mettait en place des tests ayant besoin d'un gedcom, alors on pourrait remettre un fichier grimaldi.ged, ou tout autre gedcom qu'on aurait le droit de diffuser. Ce n'est actuellement pas le cas, mais je suis ouvert à toute PR qui testerai le parsing de gedcom et la creation de base qui en découle.

dominique95880 commented:
Base grimaldi

Lorsque l'on veut faire remonter des bugs, je crois qu'il est plus efficace de tous partir du même jeu d'essai.
Le jeu d'essai parfait contiendrait toutes les situations possibles et je sais bien que cela n'a jamais existé et n'existera jamais.
La base grimaldi offre une assez grande diversité tout en restant très petite et donc facile à utiliser et à régénérer si modif.
Si elle ne contient pas le cas buggé à mettre en évidence, il est aisé d'indiquer les quelques MAJ à faire en ligne pour introduire le cas en question. (C.f. intervention de lledieu  dans un autre post).
Il faut bien sur une base "qu'on aurait le droit de diffuser". La base grimaldi me semblait avoir cette propriété (les personnages sont dans la vie publique, C.f. presse spécialisée).
Bien entendu, toute autre base peut faire consensus, l'important, c'est que l'on partage le même référentiel.

GuillaumeBrochu commented:
En effet, je trouvais aussi la base Grimaldi très utile pour les tests.

lledieu commented:
Est-il normal que pour le lieux "Roubaix, Nord, France", on me propose de modifier ? 
- Nord (ce lieu n'existe pas en tant que tel dans ma base) => m=MOD_DATA&nbs=3&data=place&s=Nor#Roubaix, Nord, France
- Roubaix => m=MOD_DATA&nbs=3&data=place&s=#Roubaix, Nord, France
Je m'attendais à trouver ce lien pour la modification : 
m=MOD_DATA&nbs=3&data=place&s=Roubaix, Nord, France

lledieu commented:
Concernant max_rlm_nbr, il va falloir généraliser.
Avec ma base, il y a 159954 inidividus pour France et ça me propose un lien qui fait des kilomètres depuis la page m=PS&bi=on&bp=on&ma=on&de=on&bu=on !
Il faut une valeur de seuil par défaut dans le code avec surcharge possible dans le fichier .gwf

hgouraud committed:
build m=L list only if <max_rlm_nbr

lledieu commented:
Au niveau de la feuille de l'arbre (patronyme), il est peut-être préférable de construire le lien à rallonge ;)

Sinon, quel est l'intérêt d'avoir dans l'arbre les lieux, des liens pour k2, k3 ? En effet pour Roubaix, ça me donne :  m=PS&bi=on&bp=on&de=on&bu=on&ma=on&dates=on&k1=France&k2=Nord&long=on

dominique95880 commented:
Mes lieux sont pour la plupart sur deux niveaux : Roubaix (59) et non Roubaix, Nord, France et k1 + k2 me suffisent (j'ai bien quelques lieux sur plus de 2 niveaux, mais dans ce cas, je n'utilise pas le lien vers le dictionnaire des lieux)

Peut-être faudrait-il prolonger k1 k2 pour avoir k1 k2 k3 ...
Quelle est la bonne limite aux kx ?
Faut-il que cela devienne illimité et avoir une procédure récursive ?

Dominique

lledieu commented:
Pour la bascule tri alpha / tri fréquence, je pense qu'il serait préférable de trouver moyen de gérer ça au niveau javascript pour éviter de relancer une requête.

lledieu commented:
Pour les histoires de liens à rallonge, ils sont intéressants car ils permettent de ne pas refaire les calculs. Je suis surpris que GeneWeb arrive a les gérer. Je n'ai pas encore testé en mode CGI, mais ça devrait poser problème car la taille de la requête est limitée en mode GET. On va probablement devoir remplacer les GET par des POST (a => form).

Il faut aussi que je teste l'ajout de l'option opt=no_index dans l'URL pour vérifier comment ça se comporte. J'ai l'impression qu'il faut prévoir un mode sans calcul et un mode avec calcul afin de pouvoir proposer un lien plus stable...

lledieu commented:
> Il faut aussi que je teste l'ajout de l'option opt=no_index dans l'URL pour vérifier comment ça se comporte.

Aïe ! opt=no_index n'a pas été prévu pour gérer plus de 10 personnes. De plus, si je remplace dans l'URL i0= par p0=&n0=&oc0=, j'obtiens ici une exception Failure("int_of_string"). Il va falloir que je regarde d'un peu plus près le code...

hgouraud commented:
Le code fait l’hypothèse i=nn

lledieu commented:
> Le code fait l’hypothèse i=nn

Dans ce cas, il est préférable d'utiliser un autre paramètre que i<n> qui a une signification particulière dans Geneweb. Je serais même d'avis de n'avoir qu'un seul paramètre qui contienne une liste d'ID séparée par des ","... Il faut être conscient qu'il ne sera pas possible d'avoir un lien stable pour la requête m=L.

lledieu commented:
Comme j'ai un peu de mal à suivre ce qu'il y a dans la branche hgouraud:place1, j'ai créé une nouvelle branche chez moi pour proposer quelques ajustements en repartant du master : 
https://github.com/geneweb/geneweb/compare/master...lledieu:place

@hgouraud : Tout nouveau filtrage doit être fait dans la fonction qu'on passe en paramètre de get_all. Ainsi les performances ne sont pas dégradées.

J'ai ajouté la préservation du lieu d'origine et j'ai fait le lien avec la recherche avancée et la modification des lieux. Finalement, c'était plus simple que ce que j'avais imaginé. Il me semble que tu peux reprendre la main dessus maintenant que le gros du traitement est fait.

Je vais tout de même essayer d'améliorer un peu la recherche avancée maintenant que j'ai compris son fonctionnement...

hgouraud commented:
Replaced by #780
